cmd/ascii2der/scanner.go

```diff
@@ -284,6 +284,51 @@ again:
 		if s.pos.Offset+1 < len(s.text) && s.text[s.pos.Offset+1] == '"' {
 			return s.parseUTF32String()
 		}
+	case 'b':
+		if s.pos.Offset+1 < len(s.text) && s.text[s.pos.Offset+1] == '`' {
+			s.advance() // Skip the b.
+			s.advance() // Skip the `.
+			bitStr, ok := s.consumeUpTo('`')
+			if !ok {
+				return token{}, &parseError{s.pos, errors.New("unmatched `")}
+			}
+
+			// Leading byte is the number of "extra" bits at the end.
+			var bitCount int
+			var sawPipe bool
+			bits := []byte{0}
+			for i, r := range bitStr {
+				switch r {
+				case '0', '1':
+					if bitCount%8 == 0 {
+						bits = append(bits, 0)
+					}
+					if r == '1' {
+						bits[len(bits)-1] |= 1 << uint(bitCount%8)
```

Comment:
I think you've got the bits backwards. Ditto with writer.go. The bits are ordered from 8 to 1 (X.690-202102, 8.6.2.1) or MSB to LSB. This is doing the LSB first.

You can see this in the `101010100101` test below. The test says {0x04, 0x??, 0x0?}, but a DER BIT STRING on a nibble boundary will always look like {0x04, 0x??, 0x?0}.

Reply:
...yup. I made sure to try it the right way around and still got it wrong. Thanks for catching this. =/